File: android/app/src/main/java/com/whoisup/app/ExtendedBranchPlugin.java
package com.whoisup.app;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

import org.json.JSONException;
import org.json.JSONObject;

import io.branch.referral.Branch;

@CapacitorPlugin(name = "ExtendedBranch")
public class ExtendedBranchPlugin extends Plugin {
    @PluginMethod
    public void getLastAttributedTouchData(PluginCall call) {
        Branch branch = Branch.getInstance();

        if (branch != null) {
            Integer attributionWindow = call.getInt("attributionWindow", 1);
            branch.getLastAttributedTouchData(
                    (jsonObject, error) -> {
                        if (error == null) {
                            if (jsonObject != null) {
                                try {
                                    JSONObject data = jsonObject.getJSONObject("last_attributed_touch_data");
                                    call.resolve(JSObject.fromJSONObject(data));
                                } catch (JSONException ignore) {
                                    // something went wrong while reading json
                                }
                            }
                            return;
                        }
                        call.resolve();
                    }, attributionWindow != null ? attributionWindow : 1);
        } else {
            call.resolve();
        }
    }
}

File: ios/App/App/ExtendedBranchPlugin.swift
import Capacitor
import BranchSDK

@objc(ExtendedBranchPlugin)
public class ExtendedBranchPlugin: CAPPlugin {
    @objc func getLastAttributedTouchData(_ call: CAPPluginCall) {
        let attributionWindow = call.getInt("attributionWindow", 1)

        DispatchQueue.main.async {
            Branch.getInstance().lastAttributedTouchData(withAttributionWindow: attributionWindow) { (result, _) in
                guard let json = result?.lastAttributedTouchJSON else {
                    call.resolve()
                    return
                }
                let js = json["data"] as? NSDictionary
                call.resolve(["data": js ?? JSObject()])
            }
        }
    }
}
